avocado/utils/dmesg_utils.py

```diff
@@ -0,0 +1,88 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2018 IBM
+# Author: Praveen K Pandey <praveen@linux.vnet.ibm.com>
+#
+
+"""
+Module for manipulate dmesg while running test.
+"""
+
+
+import time
+from . import process, genio
+
+_dmesg_log_file = '%s/dmesglog-%s' % (genio._log_file_dir,
+                                      time.strftime("%Y-%m-%d:%H:%M:%S"))
+
+
+class DmesgError(Exception):
+    """
+    Base Exception Class for all dmesg  utils exceptions
+    """
+    pass
+
+
+def clear_dmesg():
+    """
+    This function clear dmesg of system
+    and save older dmesg in a file
+    """
+
+    dmesg = collect_dmesg()
+    genio.write_file(_dmesg_log_file, dmesg)
```

Comment:
this inturn should be done in collect_dmesg

Reply:
and we can remove the dependancy of `collect_dmesg()` from `clear_dmesg()`

Reply:
@sathnaga  thanks for review , i make this as this way because i think if user clear dmesg so some time user need older dmesg  as well so thought it wil be good to before clearing dmesg we can save it some where , what you say ?

Reply:
@sathnaga ah sure that make sense if user need to save older dmesg ,user can use collect_dmesg() it return file name where it saved